tools/istio-iptables/pkg/builder/iptables-builder.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package builder
+
+// Rule represents iptables rule - chain, table and options
+type Rule struct {
+	chain  string
+	table  string
+	params []string
+}
+
+type Rules struct {
+	rulesv4 []*Rule
+	rulesv6 []*Rule
+}
+
+// IptablesProducer is an interface for adding iptables rules
+type IptablesProducer interface {
+	AppendRuleV4(chain string, table string, params ...string) IptablesProducer
```

Comment:
Please add one comment for every exported method here.

Reply:
Done

Reply:
An interface's comments shouldn't refer to implementation details not exposed in the interface, e.g. `rulesV4` here.
I'd keep it simple like:
```
	// AppendRuleV4 appends an IPv4 rule into the given iptables chain
```

Reply:
Same for the other methods below.